### PR TITLE
arm64: dts: qcom: msm8916-oppo-a51f : add initial device tree

### PR DESCRIPTION
**Pull request by Hinokami-Kagura**
for oppo-a51f with Linux kernel 5.7.1 mainline, but it can't boot, maybe somewhere need modify...XD

### CODE REVIEW

File: arch/arm64/boot/dts/qcom/msm8916-oppo-a51f.dts [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+
+			vmmc-supply = <&pm8916_l11>;
+			vqmmc-supply = <&pm8916_l12>;
+
+			pinctrl-names = "default", "sleep";
+
+			pinctrl-0 = <0xcc 0xcd 0xce 0xcf>;
+			pinctrl-1 = <0xd0 0xd1 0xd2 0xd3>;
+
+            cd-gpios = <0x48 0x26 0x00>;
```
Comment (Minecrell) [inline arch/arm64/boot/dts/qcom/msm8916-oppo-a51f.dts:44]: This looks totally broken. It should be `&msmgpio` followed by some gpio number in the example.

I recommend that you look at the downstream source, not the binary dtb: https://github.com/oppo-source/R7plus-5.1-kernel-source/blob/master/arch/arm/boot/dts/15009/msm8916-mtp.dtsi

File: arch/arm64/boot/dts/qcom/msm8916-oppo-a51f.dts [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+		etm@85c000 { status = "disabled"; };
+		etm@85d000 { status = "disabled"; };
+		etm@85e000 { status = "disabled"; };
+		etm@85f000 { status = "disabled"; };
+	};
+};
+
+// TODO (wiki): Explain setup of dummy extcon device to make USB work
+
+// FIXME: Verify all regulators with downstream
```
Comment (Minecrell) [inline arch/arm64/boot/dts/qcom/msm8916-oppo-a51f.dts:97]: remove todo, looks fine

File: arch/arm64/boot/dts/qcom/msm8916-oppo-a51f.dts [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+		debug@854000 { status = "disabled"; };
+		debug@856000 { status = "disabled"; };
+		etm@85c000 { status = "disabled"; };
+		etm@85d000 { status = "disabled"; };
+		etm@85e000 { status = "disabled"; };
+		etm@85f000 { status = "disabled"; };
+	};
+};
+
+// TODO (wiki): Explain setup of dummy extcon device to make USB work
```
Comment (Minecrell) [inline arch/arm64/boot/dts/qcom/msm8916-oppo-a51f.dts:95]: It looks like you have an usbid-gpio, please set that up similar to bq-paella:

https://github.com/msm8916-mainline/linux/blob/1d2ab0d6ec15366e9b382421630c32387a3f2157/arch/arm64/boot/dts/qcom/msm8916-bq-paella.dts#L222-L227

https://github.com/msm8916-mainline/linux/blob/1d2ab0d6ec15366e9b382421630c32387a3f2157/arch/arm64/boot/dts/qcom/msm8916-bq-paella.dts#L123-L138

File: arch/arm64/boot/dts/qcom/msm8916-oppo-a51f.dts (hunk 2 above)
Comment (Hinokami-Kagura) [reply arch/arm64/boot/dts/qcom/msm8916-oppo-a51f.dts:104]: OK, removed.

File: arch/arm64/boot/dts/qcom/msm8916-oppo-a51f.dts [hunk 4]
```diff
@@ -0,0 +1,328 @@
...
+
+			vmmc-supply = <&pm8916_l11>;
+			vqmmc-supply = <&pm8916_l12>;
+
+			pinctrl-names = "default", "sleep";
+
+	        pinctrl-0 = <&sdc2_clk_on &sdc2_cmd_on &sdc2_data_on &sdc2_cd_on>;
+	        pinctrl-1 = <&sdc2_clk_off &sdc2_cmd_off &sdc2_data_off &sdc2_cd_off>;
+
+	        cd-gpios = <&msmgpio 38 0x0>;
```
Comment (Minecrell) [inline arch/arm64/boot/dts/qcom/msm8916-oppo-a51f.dts:44]: indentation is weird here, please use tabs. Also `0x0 = GPIO_ACTIVE_HIGH`

```suggestion
	        cd-gpios = <&msmgpio 38 GPIO_ACTIVE_HIGH>;
```

see

https://github.com/msm8916-mainline/linux/blob/1d2ab0d6ec15366e9b382421630c32387a3f2157/include/dt-bindings/gpio/gpio.h#L13-L15

**Review by Minecrell — CHANGES_REQUESTED** [review]

File: arch/arm64/boot/dts/qcom/msm8916-oppo-a51f.dts [hunk 5]
```diff
@@ -0,0 +1,328 @@
...
+
+			vmmc-supply = <&pm8916_l11>;
+			vqmmc-supply = <&pm8916_l12>;
+
+			pinctrl-names = "default", "sleep";
+
+	        pinctrl-0 = <&sdc2_clk_on &sdc2_cmd_on &sdc2_data_on &sdc2_cd_on>;
+	        pinctrl-1 = <&sdc2_clk_off &sdc2_cmd_off &sdc2_data_off &sdc2_cd_off>;
+
+	        cd-gpios = <&msmgpio 38 GPIO_ACTIVE_HIGH>;
```
Comment (Minecrell) [inline arch/arm64/boot/dts/qcom/msm8916-oppo-a51f.dts:44]: there are still some spaces here instead of tabs

Comment (Hinokami-Kagura) [reply arch/arm64/boot/dts/qcom/msm8916-oppo-a51f.dts]: OK, modified!

**Comment by Minecrell** [comment]
Hi @Hinokami-Kagura. The indentation is still wrong in several places. Please make sure you use tabs consistently.
Also, please make commits with proper commit titles (see other commits in this repository as example).

The changes for the display panel should be in a separate commit later, so I would suggest that you remove them for now and finish up the base dts first. Let me know if you need any help. Thanks.

File: arch/arm64/boot/dts/qcom/msm8916-oppo-a51f.dts (hunk 5 above)
Comment (Hinokami-Kagura) [reply arch/arm64/boot/dts/qcom/msm8916-oppo-a51f.dts]: Seems I need to use terminal nano for avoiding the "auto-filling" feature by GUI editor...XD

**Review by Hinokami-Kagura — COMMENTED** [review]
removed the panel and dsi for initial only

**Pull request closed without merging**